### PR TITLE
insights: refactor- update insight view resolver to preload series data for the 1 case it didn't already

### LINKED ISSUE #27089: insights: deprecate `insights` query
Code insights has a query in GraphQL - `insights` that was the first version of interacting with code insights over API. Since then we have added new queries that have new behaviors and assumptions, that align with the product after we migrated from insights defined insettings.

Known things to change:
1. Deprecate `insights` GraphQL query
2. Deprecate filter arguments on the `InsightsSeries` GraphQL type (replaced by arguments on the `InsightViews` query)

### LINKED ISSUE #33804: insights: refactor where time series are resolved in the resolver hierarchy
Refactor where the timeseries data is loaded for insight views such that we don't have multiple disparate places loading data depending on the series type. Most likely this will mean raise it to the highest level in the resolver hierarchy (InsightViewResolver) for all types.

[Here is a notebook with a walkthrough](https://sourcegraph.com/notebooks/Tm90ZWJvb2s6NjM4#insight-series-loading-behavior-5fcdd217-2e0b-4ef0-9636-375e87260a62)

And here is a diagram showing the hierarchy and where the loading occurs for different types.
export(5)

### PR DESCRIPTION
**Pull request by chwarwick**
Previously the timeseries was loaded in the InsightViewResolver for JustInTime and Capture Group recorded series.  Move the loading of the standard search timeseries data into InsightViewResolver for consistency.

Removed unused arguments to `dataPoints` field.  

Marked legacy `insights` query as deprecated and a todo to remove a resolver that should be removed when `insights` is removed.

resolves #33804
closes #27089
## Test plan
No functionality change - existing automated tests pass

Manual testing:
Created & verified 4 insights:

1. Just in time 
2. Just in time capture group
3. backfilled 
4. backfilled capture group

### CODE REVIEW

**Comment by chwarwick** [comment]
If you review by commit you see the steps:
1 . removing the arguments that applied to only some insight types
2. mark query as deprecated
3. use the precalculated resolver to load datapoints in the insightview resolver.

File: enterprise/internal/insights/resolvers/insight_series_resolver.go [hunk 1]
```diff
@@ -32,8 +32,9 @@ import (
 	searchquery "github.com/sourcegraph/sourcegraph/internal/search/query"
 )
 
-var _ graphqlbackend.InsightSeriesResolver = &insightSeriesResolver{}
+var _ graphqlbackend.InsightSeriesResolver = &precalculatedInsightSeriesResolver{}
 
+// TODOL(insights): remove insightSeriesResolver when `insights` is removed from graphql query schema
```
Comment (CristinaBirkel) [inline enterprise/internal/insights/resolvers/insight_series_resolver.go:37]: Hmm, I don't think this is true? The `insightSeriesResolver` gets used by the `insightViewResolver` which is not going anywhere. Or maybe I'm missing something?

File: enterprise/internal/insights/resolvers/insight_series_resolver.go [hunk 2]
```diff
@@ -351,16 +323,60 @@ func expandCaptureGroupSeriesRecorded(ctx context.Context, definition types.Insi
...
+		metadataStore:   r.insightStore,
+		points:          points,
+		label:           definition.Label,
+		filters:         filters,
+		seriesId:        definition.SeriesID,
+		statusResolver:  statusResolver,
+	})
+	return resolvers, nil
+
+}
```
Comment (CristinaBirkel) [inline enterprise/internal/insights/resolvers/insight_series_resolver.go:370]: Nitpick: maybe more whitespace than needed (especially first and last line) of this function?

File: cmd/frontend/graphqlbackend/insights.go [hunk 3]
```diff
@@ -79,7 +79,7 @@ type InsightsPointsArgs struct {
 type InsightSeriesResolver interface {
 	SeriesId() string
 	Label() string
-	Points(ctx context.Context, args *InsightsPointsArgs) ([]InsightsDataPointResolver, error)
+	Points(ctx context.Context) ([]InsightsDataPointResolver, error)
```
Comment (CristinaBirkel) [inline cmd/frontend/graphqlbackend/insights.go:82]: How confident are we that this is safe to remove? The `include`/`exclude` repo options exist elsewhere, but I don't think the `to`/`from` ones are something that can be done another way, unless I'm just not thinking of it. Could be a natural fit for the `seriesDisplayOptions` I'm adding, if we need them again. But yeah just want to double check.

**Review by CristinaBirkel — APPROVED** [review]
Looks good! One minor comment, and also just want to double check that we don't actually need that to/from functionality anywhere. 

Testing wise, I made sure it loaded the four different types of insights mentioned. If you wanted to add automated API tests you could at this point, but that could also be a separate issue. They are a great idea though I think with refactors!

File: cmd/frontend/graphqlbackend/insights.go (hunk 3 above)
Comment (chwarwick) [reply cmd/frontend/graphqlbackend/insights.go:82]: This was the part that I didn't like having to do.  The comment I removed about  keeping `to preserve backwards compatibility,` was around 6 months ago, and I searched the UI code and didn't find them being used anywhere.  Currently they are ignored for 3 out of 4 instances, they only work for backend search insights.  

The alternative is that I leave the arguments in place with a note in the schema that they are always ignored.  Unfortunately the graphql spec does not support deprecation of arguments, so there isn't really a good solution.

@sourcegraph/code-insights-frontend do you know of anywhere the arguments are used when asking for `Points` 
```
(from: DateTime, to: DateTime, includeRepoRegex: String, excludeRepoRegex: String)
```
from what I can tell `from` and `to` don't apply anymore it's whatever exists or 7 points if just in time.  The include/exclude moved to the insightview.

**Comment by chwarwick** [comment]
I made a [draft of a 2nd version](https://github.com/sourcegraph/sourcegraph/pull/35014) where I left the arguments in but ignore them in all cases.  I'm open to either approach.

**Comment by chwarwick** [comment]
Closing this in favor of https://github.com/sourcegraph/sourcegraph/pull/35014 which does the same thing but leaves the `points` arguments in place, will create a new issue to remove them.

**Pull request closed without merging**